### PR TITLE
fix(python): update visionai fields default value

### PR DESCRIPTION
**Pull request by christianmasa**
## Purpose

modify default values of visionai fields except of `objects` and `contexts` related fields.

[AB#13467](https://dev.azure.com/linkerengineer/Dataverse/_sprints/taskboard/Dataverse%20Team/Dataverse/Sprint%2033%20-%20VisionAI%20Fomat%20Refinement?workitem=13467)

## What Changes?

- change fields default value
- adding validation for fields
- modify test behaviour

Mainly changes `visionai_data_format/schemas/visionai_schema.py`, `tests/test_schemas.py`, and `conftest.py` files.

## What to Check?

it should work. Will be merged to `feat/refactor-visionai-schema`, since we will update `contexts` and `objects` related in the next PR

### CODE REVIEW

File: visionai_data_format/schemas/visionai_schema.py [hunk 1]
```diff
@@ -558,18 +589,43 @@ class Config:
...
+        return value
+
+
 class CoordinateSystem(BaseModel):
-    type: StrictStr
-    parent: StrictStr = ""
-    children: List[StrictStr] = Field(default_factory=list)
+    type: CoordinateSystemType
+    parent: StrictStr
+    children: List[StrictStr] = Field(...)
```
Comment (JasperSui) [inline visionai_data_format/schemas/visionai_schema.py:606]: Can remove the `default=...`.

File: visionai_data_format/schemas/visionai_schema.py [hunk 2]
```diff
@@ -78,7 +85,7 @@ class Config:
 
 class ObjectDataElement(BaseModel):
 
-    attributes: Attributes = Field(default_factory=dict)
+    attributes: Optional[Attributes] = Field(default=None)
```
[the remark this answers is not in the document]
Comment (christianmasa) [reply visionai_data_format/schemas/visionai_schema.py:88]: will revert this change for now.

File: visionai_data_format/schemas/visionai_schema.py [hunk 3]
```diff
@@ -581,44 +637,76 @@ class VisionAI(BaseModel):
...
     )
 
+    @validator("coordinate_systems")
+    def validate_coordinate_systems(cls, value):
+        assert value, f" Value {value} is not allowed"
+        return value
+
     streams: Dict[StrictStr, Stream] = Field(
-        default_factory=dict,
+        default=...,
```
[the remark this answers is not in the document]
Comment (christianmasa) [reply visionai_data_format/schemas/visionai_schema.py]: done

File: visionai_data_format/schemas/visionai_schema.py [hunk 4]
```diff
@@ -581,44 +637,76 @@ class VisionAI(BaseModel):
...
         description="This is the JSON object of VisionAI that contains the streams and their details.",
     )
 
-    metadata: Metadata = Field(default_factory=Metadata)
+    metadata: Metadata = Field(default=...)
```
[the remark this answers is not in the document]
Comment (christianmasa) [reply visionai_data_format/schemas/visionai_schema.py]: done

File: visionai_data_format/schemas/visionai_schema.py (hunk 1 above)
Comment (christianmasa) [reply visionai_data_format/schemas/visionai_schema.py]: done

File: visionai_data_format/schemas/visionai_schema.py [hunk 5]
```diff
@@ -451,19 +479,22 @@ class TimeStampElement(BaseModel):
...
-
 
 class FramePropertyStream(BaseModel):
     uri: str = Field(..., description="the urls of image")
-    stream_properties: Optional[
-        Dict[StrictStr, StreamPropertyUnderFrameProperty]
-    ] = Field(None, description="Additional properties of the stream")
+    stream_properties: Optional[StreamPropertyUnderFrameProperty] = Field(
+        None, description="Additional properties of the stream"
+    )
```
[the remark this answers is not in the document]
Comment (christianmasa) [reply visionai_data_format/schemas/visionai_schema.py:507]: added

File: visionai_data_format/schemas/visionai_schema.py [hunk 6]
```diff
@@ -687,14 +700,19 @@ def validate_objects(cls, value):
     @validator("coordinate_systems")
     def validate_coordinate_systems(cls, value):
         assert value, f" Value {value} is not allowed"
+
+        for k, v in value.items():
+            if v.type == "local_cs" and "iso8855" not in k:
+                raise ValueError(
+                    f"Can't assign coordinate system {k} with local_cs type"
+                )
         return value
```
Comment (christianmasa) [inline visionai_data_format/schemas/visionai_schema.py]: validate only `iso8855` related keys that could have `local_cs` type

File: visionai_data_format/schemas/visionai_schema.py [hunk 7]
```diff
@@ -396,20 +393,29 @@ class Config:
...
+    def validate_distortion_coeffs_1xN(cls, value):
+        assert value, f"Value {value} is not allowed"
+        return value
 
     @validator("camera_matrix_3x4")
     def validate_camera_matrix_3x4(cls, value):
         assert (
-            len(value) == 16
-        ), f"Value {value} with length {len(value)} is not allowed"
+            value and len(value) == 12
```
Comment (christianmasa) [inline visionai_data_format/schemas/visionai_schema.py:410]: wrong validation,  camera matrix with 3x4 contains only 12 value

**Review by JasperSui — APPROVED** [review]
LGTM!

**Comment by christianmasa** [comment]
will be merged for now

**Pull request merged**